src/screens/nano-contract/NanoContractDetail.js

```diff
@@ -51,25 +53,57 @@ function NanoContractDetail() {
   const navigate = useNavigate();
   const { nc_id: ncId } = useParams();
   const nc = nanoContracts[ncId];
-  let blueprintInformationAux = nc != null ? blueprintsData[nc.blueprintId] : null;
 
   // loading {boolean} Bool to show/hide loading element
   const [loading, setLoading] = useState(true);
   // data {Object} Nano contract loaded data
   const [data, setData] = useState(null);
   // blueprintInformation {Object} Blueprint information data
-  const [blueprintInformation, setBlueprintInformation] = useState(blueprintInformationAux);
+  const [blueprintInformation, setBlueprintInformation] = useState(null);
   // errorMessage {string} Message to show when error happens on the form
   const [errorMessage, setErrorMessage] = useState('');
+  // waitingConfirmation {boolean} If transaction was loading and is waiting first block confirmation
+  const [waitingConfirmation, setWaitingConfirmation] = useState(false);
 
   useEffect(() => {
     if (nc) {
       // Load data only if nano contract exists in redux,
       // otherwise it has just been unregistered
-      loadData();
+      dispatch(nanoContractDetailRequest(ncId));
+    }
+
+    return () => {
+      // Move state to ready when unmounting
+      dispatch(nanoContractDetailSetStatus({ status: NANO_CONTRACT_DETAIL_STATUS.READY }));
     }
   }, []);
 
+  useEffect(() => {
+    if (nanoContractDetailState.status === NANO_CONTRACT_DETAIL_STATUS.LOADING) {
+      setLoading(true);
+    }
+
+    if (nanoContractDetailState.status === NANO_CONTRACT_DETAIL_STATUS.ERROR) {
+      setLoading(false);
+      setErrorMessage(nanoContractDetailState.error);
+    }
```

Comment:
Do you really need this? 

Can't you just use `nanoContractDetailsState` directly when rendering?

Reply:
I think we should always avoid having stateful components

The state is our enemy